### ISSUE TITLE
[FR] "@file:Suppress" on wrong location triggers unclear "Annotations should occur immediately before the annotated construct"

### ISSUE THREAD

**Issue by Zocker1999NET** [pinterest/ktlint#3334]
I'm new to Kotlin, so I'm not used which is the expected order of statements on top of new Kotlin files.

Context was, I wanted to add a `@file:Suppress("…")` statement to a file, but that triggered an ktlint error stating "Annotations should occur immediately before the annotated construct" added to the huge list of other issues which should be suppressed by this annotation. So it took me a long time until I understood that:
- the `@file:Suppress` was at the wrong location
- that because of this ktlint effectively ignored this suppression
- and where the correct location could be.

My (invalid) code was:
```kotlin
package com.mednotify

import com.tngtech.archunit.junit.AnalyzeClasses
import com.tngtech.archunit.junit.ArchTest
import com.tngtech.archunit.junit.ArchUnitRunner
import com.tngtech.archunit.lang.ArchRule
import com.tngtech.archunit.lang.syntax.ArchRuleDefinition
import org.junit.runner.RunWith

// because readable package name variables
// because readable rule names
@file:Suppress("ktlint:standard:property-naming")

// for readability, consts not in uppercase
private const val AppPackage = "example.mednotify" // TODO update
private const val DataLayer = "$AppPackage.data.."
private const val DomainLayer = "$AppPackage.domain.."
private const val NotificationPkg = "$AppPackage.notification.."
private const val TriggerPkg = "$AppPackage.trigger.."
private const val UILayer = "$AppPackage.ui.."

```

After trying out, I discovered that `@file:Suppress` statements need to be placed at the top before the `package` statement. Makes sense after thinking about, but this was not my first idea because this was the first time using something which needs to be placed before `package`, and because in other languages & linters, file suppressions can be placed everywhere in the file.

If possible, I propose following changes:
- if the given annotation is a (probably) suppression for ktlint, change the message to state that other this probably should to be fixed first because the suppression is not effective
- if the given annotation is a file-level suppression, change the message to state that the annotation needs to appear at the top of the file before the `package` statement.

This would make it easy for beginners, esp. because it took me long to figure out that those suppression annotations are a Kotlin wide feature & most probably because of that, ktlint does not document themselves how those are used, hence I did not found there how file level suppressions are to be used.

**Comment by paul-dingemans** [comment]
Hi @Zocker1999NET,

Thanks for yur suggestion. I do not think that Ktlint should do this. The compiler warns about incorrect usage of the `file` annotation. I would not advise to run ktlint as long as you need to resolve compilation errors.

<img width="1052" height="559" alt="Image" src="https://github.com/user-attachments/assets/25ef59dd-2d0d-416d-9165-d58a14562940" />

**Comment by Zocker1999NET** [comment]
Sorry for my late answer & sorry that I forgot to mention the context of my ktlint usage.

The compiler does not (yet) show this as an error when using ktlint as Gradle plugin. Because then, ktlint is executed before the compile tasks are & therefore the message I quoted before was everything I had. I'm not sure whether this is due to an actual dependency or because ktlint is just completed quicker compared to the compiler.

I'm currently using ktlint by integrating it as follows into my `build.grade.kts`:
```kotlin
plugins {
    …
    alias(libs.plugins.ktlint)
    …
}

ktlint {
    version = "1.8.0"
    android = true
}

…
```

Because of other reasons, I'm currently still forced for now to use an IDE without Kotlin support, hence I do not yet see any errors from the compiler before executing Gradle manually. And the same would be true when working with e.g. the output from a CI/CD pipeline.

I can understand if you still think its not worth to do anything about this, through I still think at least a little adaption of the message would make it way easier for Kotlin beginners to start working on an existing or derived project where ktlint is integrated into Gradle.

**Comment by paul-dingemans** [comment]
Thanks for adding this context. Now I do understand your viewpoint better. Despite that, Ktlint Core will not be changed. There will be more situations in which Ktlint violations might not make sense as long as the code contains compilation errors.
